### PR TITLE
fix flakly ingressnginx test

### PR DESCRIPTION
**Pull request by LiorLieberman**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Ingress-nginx support mergind different ingresses to one HTTPRoute, but since we used a map to store the ingresses, the order is not consistent, causing tests to be flaky
**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fixed ingress-nginx conversion tests
```

/cc @levikobi 
FYI @YTGhost

### CODE REVIEW

File: pkg/i2gw/providers/ingressnginx/resource_reader.go [hunk 1]
```diff
@@ -42,7 +44,14 @@ func (r *resourceReader) readResourcesFromCluster(ctx context.Context) (*storage
...
 	}
-	storage.Ingresses = ingresses
+	ingNames := []types.NamespacedName{}
+	for ing := range ingresses {
+		ingNames = append(ingNames, ing)
+	}
+	sort.Slice(ingNames, func(i, j int) bool {
+		return ingNames[i].Name < ingNames[j].Name
+	})
+	storage.Ingresses = OrderedIngressMap{ingressNames: ingNames, ingressObjects: ingresses}
```
Comment (levikobi) [inline pkg/i2gw/providers/ingressnginx/resource_reader.go:54]: I think having this functionality as a method/constructor of the `OrderedIngressMap` type will make things easier to maintain, especially as we already duplicate it later in this file.

File: pkg/i2gw/providers/ingressnginx/converter.go [hunk 2]
```diff
@@ -41,8 +41,8 @@ func (c *converter) convert(storage *storage) (i2gw.GatewayResources, field.Erro
 
 	// TODO(liorliberman) temporary until we decide to change ToGateway and featureParsers to get a map of [types.NamespacedName]*networkingv1.Ingress instead of a list
 	ingressList := []networkingv1.Ingress{}
-	for _, ing := range storage.Ingresses {
-		ingressList = append(ingressList, *ing)
+	for _, ing := range storage.Ingresses.ingressNames {
+		ingressList = append(ingressList, *storage.Ingresses.ingressObjects[ing])
```
Comment (levikobi) [inline pkg/i2gw/providers/ingressnginx/converter.go:45]: Consider creating a method on the `OrderedIngressMap` (something like `OrderedIngressMap.Ingresses()`) that gives you the ordered list of ingresses, to abstract away the fact that we must access the `ingressObjects` using the ordered `ingressNames`

**Review by levikobi — COMMENTED** [review]
Overall looks good to me, I have some suggestions surrounding the maintainability of this addition

File: pkg/i2gw/providers/ingressnginx/resource_reader.go (hunk 1 above)
Comment (LiorLieberman) [reply pkg/i2gw/providers/ingressnginx/resource_reader.go]: Done, Thanks!

**Review by levikobi — COMMENTED** [review]
/lgtm

**Pull request merged**